### PR TITLE
github/docs: Add a Pull Request Template

### PR DESCRIPTION
**Pull request by xTVaser**
A small file move cleanup, and created the only template we are currently missing -- a pull-request one.  Kept separate from https://github.com/PCSX2/pcsx2/pull/4327 incase we don't want this.

I tend to keep the PR templates light because in my experience either the majority of it gets cumbersome/ignored or isn't applicable to every PR.

### CODE REVIEW

**Comment by lightningterror** [comment]
I actually was planning in adding this, can you also add info such as:
properly naming commits along with prefixes,
keeping formatting changes in separate commits,
following/running clang if applicable on the newer changes,
for members of the organizations to also mark their prs with labels based on the changes.

**Comment by xTVaser** [comment]
Can we update the markdown linter's filename regex to `\\.a-zA-Z0-9-_`  

Should be able to find it under `Code Patterns` > `Remark Linter`:
image

**Comment by lightningterror** [comment]
Done.

**Comment by xTVaser** [comment]
> for members of the organizations to also mark their prs with labels based on the changes.

Handled automatically here - https://github.com/PCSX2/pcsx2/pull/4332

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - Try to keep messages brief and informative
+  - Remove unnecessary commits and squash commits together when appropriate.
+    - If you are not familiar with rebasing, check out the following resources:
+      - CLI - https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history
+      - GUI (SourceTree) - https://www.atlassian.com/blog/sourcetree/interactive-rebase-sourcetree
+- Code Styling and Formatting
+  - Consult the style guide - https://github.com/tadanokojin/pcsx2/blob/coding-guide/pcsx2/Docs/Coding_Guidelines.md
+  - Run `clang-format` using the configuration file in the root of the repository
+    - Visual Studio Setup - https://devblogs.microsoft.com/cppblog/clangformat-support-in-visual-studio-2017-15-7-preview-1/
+    - IMPORTANT - if you are running `clang-format` on unrelated changes (ie. formatting an entire file), please do so in a separate commit.
```
Comment (lightningterror) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Maybe we can add if the file hasn't been clang formatted and would like to skip the formatting stick to how the rest of the file is formatted.

**Comment by lightningterror** [comment]
Can you post a demo on your repo so I can check it out?

**Comment by xTVaser** [comment]
@lightningterror Pushed to my fork's master and test made https://github.com/xTVaser/pcsx2-rr/pull/76 feel free to make a PR of your own to test it as well.

**Review by lightningterror — APPROVED** [review]

**Pull request merged**